deployment/overlays/memtierd/daemonset.yaml

```diff
@@ -53,10 +57,14 @@ spec:
         hostPath:
           path: /var/run/nri
           type: Directory
-      - name: host-vol
+      - name: run-dir-vol
```

Comment:
Why is this a host volume/bind mount ? To save data/avoid losing state across potential restarts ?

Reply:
It exposes `memtierd.output` that enables, for instance, reading results in the `track-working-set-size` class (or use case).

It could be useful also to protect against restarts due to containing memtierd.pid's. That would enable killing memtierd processes even after nri-memtierd would have crashed/restarted. However, this is not implemented.

Reply:
Should/could we make this optional (i.e. remove from kustomize and behind a parameter in Helm)?

Reply:
Or perhaps use emptyDir volume if we want persisent storage over the lifetime of the pod

Reply:
I suppose that would be a good and safe default.

Reply:
> Is there something currently in that memtierd.output which would be difficult/computationally heavy to calculate/produce on demand, because for instance some context is already lost, etc. ? If not, then an alternative would be to provide an HTTP endpoint in memtierd for querying such data.

In the track-working-set-size example memtierd.output contains full history of working set size reports, each of them being a snapshot from last access times to data in memory.

In any case, I really like the idea of having an http endpoint to serve the data. Because the output is produced by each memtierd separately, we could actually use an emptyDir to store the data from nri-memtierd container, and run our http endpoint in an unprivileged container: it can read the emptyDir and serve the data from there.

Still hesitating a bit to go this way, because right now it feels like k8s deployment would become an integral part of the implementation and architecture...

Reply:
We can cache data from a reasonably big time window into RAM or storage. Between these two I'd prefer storage because speed is not an issue and the storage gives bigger time windows with lower costs.

If we'd go with caching to storage, one reasonably simple solution could be using the `fallocate()` syscall on memtierd.output files: we simply push a bigger and bigger hole in the beginning of these files (thus deallocating first blocks), and we would add an http endpoint to the nri-memtierd container to serve the rest of the data. That would keep RAM and storage usage limited, and it would allow us to keep redirecting memtierd process output directly into a file without any reader/writer threads or any other data gathering logic. Though this may sound a bit hackish, too. @klihub, @marquiz, what do you think?

Reply:
I don't know if it's really worth the effort/makes sense to roll anything home-grown for file-based storage. The big question for me was whether file based storage could be avoided altogether. Since looks like it cannot be, I am happy with whatever you think works best.

Reply:
I definitely agree with you that this is worth addressing in the default deployment yamls in the future. For the first release I'd consider current deployments to be good enough for trying out the nri-memtierd plugin in action. The concern regarding (host) disk space is definitely valid but unfortunately not the biggest one when thinking of memtierd in production use.